addons/purchase/models/purchase.py

```diff
@@ -658,18 +658,15 @@ def _send_reminder_mail(self, send_single=False):
                     order.with_context(is_reminder=True).message_post_with_template(template.id, email_layout_xmlid="mail.mail_notification_paynow", composition_mode='comment')
 
     def send_reminder_preview(self):
-        if not self.user_has_groups('purchase.group_send_reminder') and not self.receipt_reminder_email:
-            return
-
+        self.ensure_one()
         template = self.env.ref('purchase.email_template_edi_purchase_reminder', raise_if_not_found=False)
-        if template and self.env.user.email:
-            for order in self:
-                template.with_context(is_reminder=True).send_mail(
-                    order.id,
-                    force_send=True,
-                    raise_exception=False,
-                    email_values={'email_to': self.env.user.email, 'recipient_ids': []},
-                    notif_layout="mail.mail_notification_paynow")
+        if template and self.env.user.email and self.id:
```

Comment:
why do you need to test `self.id` and calling self.ensure_one() ?

Reply:
I test self.id to decide if the PO is saved to be a record or not. Maybe there is a better way?

I add ensure_one() because this is only called by a button on the form view, I don't think it need to be adapt to a recordset.